### PR TITLE
AUDIO: Support building with FluidLite

### PR DESCRIPTION
**Pull request by ccawley2011**

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -1123,6 +1128,9 @@ for ac_option in $@; do
 	--disable-faad)               _faad=no               ;;
 	--enable-faad)                _faad=yes              ;;
 	--disable-fluidsynth)         _fluidsynth=no         ;;
+	--enable-fluidsynth)          _fluidsynth=yes        ;;
+	--disable-fluidlite)          _fluidlite=no          ;;
+	--enable-fluidsynth)          _fluidlite=yes         ;;
 	--enable-readline)            _readline=yes          ;;
```
Comment (lotharsm) [inline configure:1134]: Shouldn't this read `--enable-fluidlite` instead?

Comment (ccawley2011) [reply configure:1134]: Fixed.

File: configure [hunk 2]
```diff
@@ -1123,6 +1128,9 @@ for ac_option in $@; do
 	--disable-faad)               _faad=no               ;;
 	--enable-faad)                _faad=yes              ;;
 	--disable-fluidsynth)         _fluidsynth=no         ;;
+	--enable-fluidsynth)          _fluidsynth=yes        ;;
+	--disable-fluidlite)          _fluidlite=no          ;;
+	--enable-fluidlite)           _fluidlite=yes         ;;
```
Comment (mgerhardy) [inline configure:1133]: There is this in `gui/module.mk`

```
ifdef USE_FLUIDSYNTH
MODULE_OBJS += \
	fluidsynth-dialog.o
endif
```
Is that maybe needed for `fluidlite`, too?

**Comment by mgerhardy** [comment]
Changes to `base/version.cpp` are missing. Also the xcode generator of `create_project` has a `fluidsynth` special handling - not sure if this is needed for this lib, too.

And last but not least there is `base/plugins.cpp` which should be adopted for `fluidlite`.

File: configure (hunk 2 above)
Comment (ccawley2011) [reply configure:1133]: ``USE_FLUIDSYNTH`` is defined when fluidlite is enabled, so that shouldn't be necessary.

Comment (mgerhardy) [reply configure:1133]: ah - missed the part where you also define USE_FLUIDSYNTH in config.mk - sorry.

File: configure [hunk 3]
```diff
@@ -4966,6 +4979,38 @@ fi
...
+echocheck "FluidLite"
+
+append_var FLUIDLITE_LIBS "-lfluidlite"
+if test "$_fluidlite" = auto; then
+	_fluidlite=no
+	cat > $TMPC << EOF
+#include <fluidlite.h>
+int main(void) { delete_fluid_settings(new_fluid_settings()); return 0; }
+EOF
+	cc_check_no_clean $FLUIDLITE_CFLAGS $FLUIDLITE_CFLAGS && _fluidlite=yes
```
Comment (AReim1982) [inline configure:4994]: Shouldn't it be like this: "cc_check_no_clean $FLUIDLITE_CFLAGS $FLUIDLITE_LIBS && _fluidlite=yes"

**Review by AReim1982 — APPROVED** [review]
Looks good to me.

**Review by bluegr — APPROVED** [review]

**Comment by bluegr** [comment]
Looks good to me too

**Comment by bluegr** [comment]
@ccawley2011 could you please check if there are any changes needed for this to create_project?

File: configure [hunk 4]
```diff
@@ -4966,6 +4979,38 @@ fi
 define_in_config_if_yes "$_fluidsynth" 'USE_FLUIDSYNTH'
 echo "$_fluidsynth"
 
+#
+# Check for FluidLite
+#
+echocheck "FluidLite"
```
Comment (sev-) [inline configure:4985]: What will happen if both fluidsynth and fluidlite are present?

Comment (ccawley2011) [reply configure:4985]: In that case, only FluidSynth will be enabled.

**Comment by SupSuper** [comment]
@bluegr Added FluidLite to create_project

**Comment by bluegr** [comment]
@SupSuper thanks! That was the last missing piece

This can be merged now

**Pull request merged**